src/notification.cpp

```diff
@@ -1793,6 +1851,57 @@ QVariant Notification::remoteAction(const QString &name, const QString &displayN
     return action;
 }
 
+/*!
+    \fn Notification::actionSetDBusCall(QVariant &, const QString &, const QString &, const QString &, const QString &, const QVariantList &)
+
+    Helper function to add details to a remote action to be invoked via D-Bus. 
+
+    \list
+    \li \a action: QVariantMap created by Notification::remoteAction
+    \li \a service: the name of the D-Bus service to be invoked
+    \li \a path: the object path to be invoked via D-Bus
+    \li \a iface: the interface to be invoked via D-Bus
+    \li \a method: the method of the interface to be invoked via D-Bus
+    \li \a arguments: the optional arguments to be passed to the method invoked via D-Bus
+    \endlist
+*/
+QVariant Notification::actionSetDBusCall(QVariant &action, const QString &service, const QString &path, const QString &iface,
+                const QString &method, const QVariantList &arguments)
+{
+    QVariantMap vm = action.value<QVariantMap>();
+    vm.insert(QStringLiteral("service"), service);
+    vm.insert(QStringLiteral("path"), path);
+    vm.insert(QStringLiteral("iface"), iface);
+    vm.insert(QStringLiteral("method"), method);
+    if (!arguments.isEmpty()) {
+        vm.insert(QStringLiteral("arguments"), arguments);
+    }
+    return vm;
+}
+
+/*!
+    \fn Notification::actionSetReplyFormat(QVariant &, QString &, bool, const QStringList &)
```

Comment:
So I'm not 100% sure about adding new action API in this form. Would like to revisit the whole thing with something more explicit NotificationAction type of class. If that happens, these two new ones might end up as soon obsolete API we might have to still support.

Could be worth considering if these were either done on the app side, or then marked \internal  for now?